hwangJeongSoo/app.js

```diff
@@ -0,0 +1,69 @@
+const http = require("http");
+const express = require("express");
+const cors = require("cors");
+const morgan = require("morgan");
+
+const dotenv = require("dotenv");
+dotenv.config()
+
+const { DataSource } = require('typeorm')
+
+const myDataSource = new DataSource({
+    type : process.env.TYPEORM_CONNECTION,
+    host : process.env.TYPEORM_HOST,
+    port : process.env.TYPEORM_PORT,
+    username : process.env.TYPEORM_USERNAME,
+    password : process.env.TYPEORM_PASSWORD,
+    database : process.env.TYPEORM_DATABASE 
+})
+
+myDataSource.initialize() 
+    .then(()=>{
+        console.log("Data Source has been initialized")
+    })
+
+const app = express();
+
+app.use(express.json())
+app.use(cors())
+app.use(morgan('dev'))
+
+app.get("/ping", (req, res) =>{
+	res.status(201).json({ message : "pong"});
+    });
+//CRUD
+app.post("/signup", async(req, res, next)=>{
+    const { id, name, email, profile_image, password } = req.body
+
+
+await myDataSource.query(
+    `INSERT INTO users(
+        id,
+        name,
+        email,
+        profile_image,
+        password
+    ) VALUES (?, ?, ?, ?, ?);
+    `,
+    [ id, name, email, profile_image, password ]
+); 
```

Comment:
```suggestion
await myDataSource.query(
    `INSERT INTO users(
        name,
        email,
        profile_image,
        password
    ) VALUES (?, ?, ?, ?, ?);
    `,
    [ name, email, profile_image, password ]
); 
```
데이터베이스 users 테이블 스키마를 보면 id 컬럼은 `auto_increment` 속성이 지정되어 있네요!

데이터가 생성될 때마다 해당 데이터에 맞는 id 값이 자동으로 생성되기 때문에 직접 값을 넣어주지 않아도 됩니다!

Reply:
.